### ISSUE TITLE
About list-wise loss calculation

### ISSUE THREAD

**Issue by michaeltian108** [theFool32/UAP_retrieval#6]
Hey! I've been trying to redo your paper and I've successfully reproduced all data except list-wise noise on R-Gem. In the paper you mentioned you used DCG as  the metric for list-wise relationship. 
`            ndcg_diffs = (1 / (1 + doc_ranks[pos_i])).log2() - (
                1 / (1 + doc_ranks[neg_i])
            ).log2().view(neg_i.size(0))
`
Here you  take the log of 1/(1+i) and 1/(1+j) to calculation the dcg difference, which is quite different with the formula you use in your paper, where is defined as:
DCG =  Σ (2^y_i -1)/log2(i+1) 

Did you define  y_i as 1 for all results in the retrieved list?( Regarding them all as positive pairs?)
But even that dcg should be
`1 /torch.log2(1 + doc_ranks[pos_i])` instead of what's in your code.
Could you please explain this?

I'm looking into this because the data I got from list-wise attack is way worse than those in your paper.
Here's the command I used for training:
`python3 -m cirtorch.examples.attack.attack --network-path Output_network/retrievalSfM120k-resnet101-gem.pth --noise-path universal_advnoise/list-wise --list_wise`

Thanks!

**Comment by theFool32** [comment]
Hi, thank you for your interest.

The NDCG implemented in our code is a simplified version while we treat the top half of the rank as positive pairs (label 1) and another half as negative (label -1).

For the result of list-wise on R-Gem, can you share the results you got? 
Since it has been more than two years, I need some time to check it.

**Comment by michaeltian108** [comment]
> Hi, thank you for your interest.
> 
> The NDCG implemented in our code is a simplified version while we treat the top half of the rank as positive pairs (label 1) and another half as negative (label -1).
> 
> For the result of list-wise on R-Gem, can you share the results you got? Since it has been more than two years, I need some time to check it.


Sure things. Thanks a lot for your quick reply, and please take your time as I've also been looking into your code to see if I can work it out.

Here's the results:

List-wise best noise
Best
>> Evaluating network on test datasets...
>> oxford5k: Extracting...
>> oxford5k: database images...
>> oxford5k: query images...
>>>> 55/55 done...	
>> oxford5k: Evaluating...
>> oxford5k + whiten: mAP 53.43
>> oxford5k: elapsed time: 4s
>> paris6k: Extracting...
>> paris6k: database images...
>> paris6k: query images...
>>>> 55/55 done...
>> paris6k: Evaluating...
>> paris6k + whiten: mAP 60.81
>> paris6k: elapsed time: 4s
>> roxford5k: Extracting...
>> roxford5k: database images...
>> roxford5k: query images...
>>>> 70/70 done...
>> roxford5k: Evaluating...
>> roxford5k + whiten: mAP E: 44.81, M: 35.65, H: 16.79
>> roxford5k + whiten: mP@k[1, 5, 10] E: [50.   51.32 49.72], M: [50.   50.1  49.67], H: [28.57 27.29 25.71]
>> roxford5k: elapsed time: 4s
>> rparis6k: Extracting...
>> rparis6k: database images...
>> rparis6k: query images...
>>>> 70/70 done...
>> rparis6k: Evaluating...
>> rparis6k + whiten: mAP E: 56.3, M: 45.6, H: 21.13
>> rparis6k + whiten: mP@k[1, 5, 10] E: [40.   58.86 60.  ], M: [40.   59.43 61.29], H: [28.57 37.71 38.71]
>> rparis6k: elapsed time: 5s
